versioned/persist/dynamodb/src/main/java/org/projectnessie/versioned/persist/dynamodb/DynamoDatabaseAdapter.java

```diff
@@ -468,4 +503,101 @@ protected RefLog fetchFromRefLog(NonTransactionalOperationContext ctx, Hash refL
       NonTransactionalOperationContext ctx, List<Hash> hashes) {
     return fetchPageResult(TABLE_REF_LOG, hashes, ProtoSerialization::protoToRefLog);
   }
+
+  @Override
+  protected void writeAttachments(Stream<Entry<AttachmentKey, AttachmentValue>> attachments) {
+    batchWrite(
+        TABLE_ATTACHMENTS,
+        attachments.collect(Collectors.toList()),
+        e -> attachmentKeyAsString(e.getKey()),
+        e -> e.getValue().toByteArray(),
+        e -> {
+          AttachmentValue v = e.getValue();
+          if (!v.hasVersion()) {
+            return Collections.emptyMap();
+          }
+          return Collections.singletonMap(
+              VERSION_NAME, AttributeValue.builder().s(v.getVersion()).build());
+        });
+  }
+
+  @Override
+  protected boolean consistentWriteAttachment(
+      AttachmentKey key, AttachmentValue value, Optional<String> expectedVersion) {
+
+    AttributeValue newValueBytes =
+        AttributeValue.builder().b(SdkBytes.fromByteArray(value.toByteArray())).build();
```

Comment:
why the `AttributeValue.builder` has those not descriptive methods like `s()` or `b()`?
Can we change those to some meaningful names?

Reply:
Ehm - no? That's AWS SDK